pkg/tools/cli/loadtest/service/search.go

```diff
@@ -17,26 +17,61 @@ package service
 
 import (
 	"context"
+	"sync"
 
+	"github.com/vdaas/vald/apis/grpc/agent/core"
 	"github.com/vdaas/vald/apis/grpc/gateway/vald"
 	"github.com/vdaas/vald/apis/grpc/payload"
+	"github.com/vdaas/vald/internal/errors"
 	"github.com/vdaas/vald/internal/net/grpc"
 	"github.com/vdaas/vald/pkg/tools/cli/loadtest/assets"
+	"github.com/vdaas/vald/pkg/tools/cli/loadtest/config"
 )
 
-func newSearch() (requestFunc, loaderFunc) {
-	return func(dataset assets.Dataset) ([]interface{}, error) {
-			vectors := dataset.Query()
-			requests := make([]interface{}, len(vectors))
-			for j, v := range vectors {
-				requests[j] = &payload.Search_Request{
-					Vector: v,
-				}
+func searchRequestProvider(dataset assets.Dataset) (func() interface{}, int, error) {
+	v := dataset.Query()
+	size := len(v)
+	i := 0
+	m := sync.Mutex{}
```

Comment:
I found two pattern about creating sync.Mutex object (`sync.Mutex{} or &sync.Mutex{}`)
Could you please fit `sync.Mutex` or `&sync.Mutex{}`??

https://github.com/vdaas/vald/pull/552/files?file-filters%5B%5D=.go#diff-2141299daa3efe6326b530b7cde72e6eR51

Reply:
Thank you for your comment!
I'll fix it with the others.